centreon/tests/e2e/cypress/e2e/Dashboards/06-dashboard-sharing/index.ts

```diff
@@ -0,0 +1,797 @@
+  () => {
+    cy.loginByTypeOfUser({
+      jsonName: dashboardAdministratorUser.login,
+      loginViaApi: false
+    });
+    cy.visit(`${Cypress.config().baseUrl}/centreon/home/dashboards`);
+    cy.getByLabel({
+      label: 'view',
+      tag: 'button'
+    })
+      .contains(dashboards.fromDashboardAdministratorUser.name)
+      .click();
+    cy.getByLabel({ label: 'share', tag: 'button' }).click();
+  }
+);
+
+When('the admin user appoints one of the users as an editor', () => {
+  cy.getByLabel({ label: 'Open', tag: 'button' }).click();
+  cy.contains(dashboardCreatorUser.login).click();
+  cy.getByTestId({ testId: 'add' }).should('be.enabled');
+  cy.getByTestId({ testId: 'role-input' }).eq(0).click();
+  cy.get('[role="listbox"]').contains('editor').click();
+  cy.getByTestId({ testId: 'add' }).click();
+  cy.getByLabel({ label: 'Update', tag: 'button' }).click();
+  cy.reload();
+});
+
+Then(
+  'the newly appointed editor user can appoint another user as an editor',
+  () => {
+    cy.visit(`${Cypress.config().baseUrl}/centreon/home/dashboards`);
+    cy.logout();
+    cy.getByLabel({ label: 'Alias', tag: 'input' }).should('exist');
+
+    cy.loginByTypeOfUser({
+      jsonName: dashboardCreatorUser.login,
+      loginViaApi: false
+    });
+    cy.visit(`${Cypress.config().baseUrl}/centreon/home/dashboards`);
+    cy.getByLabel({
+      label: 'view',
+      tag: 'button'
+    })
+      .contains(dashboards.fromDashboardAdministratorUser.name)
+      .click();
+    cy.getByLabel({ label: 'share', tag: 'button' }).click();
+    cy.getByLabel({ label: 'Open', tag: 'button' }).click();
+    cy.contains(dashboardCGMember3.login).click();
+    cy.getByTestId({ testId: 'add' }).should('be.enabled');
+    cy.getByTestId({ testId: 'role-input' }).eq(0).click();
+    cy.get('[role="listbox"]').contains('editor').click();
+    cy.getByTestId({ testId: 'add' }).click();
+
+    cy.getByLabel({ label: 'Update', tag: 'button' }).click();
+    cy.reload();
+
+    cy.getByLabel({ label: 'share', tag: 'button' }).click();
+    cy.get('*[class^="MuiList-root"]')
+      .eq(1)
+      .children()
+      .contains(dashboardCGMember3.login)
+      .should('exist');
+
+    cy.get('[data-state="added"]').should('not.exist');
+    cy.getByLabel({ label: 'Cancel', tag: 'button' }).click();
+  }
+);
+
+Then(
+  'the newly appointed editor user can appoint another user as a viewer',
+  () => {
+    cy.getByLabel({ label: 'share', tag: 'button' }).click();
+    cy.getByLabel({ label: 'Open', tag: 'button' }).click();
+    cy.contains(dashboardViewerUser.login).click();
+    cy.getByTestId({ testId: 'add' }).should('be.enabled');
+    cy.getByTestId({ testId: 'role-input' }).eq(0).click();
+    cy.get('[role="listbox"]').contains('viewer').click();
+    cy.getByTestId({ testId: 'add' }).click();
+
+    cy.getByLabel({ label: 'Update', tag: 'button' }).click();
+    cy.reload();
```

Comment:
is it possible to avoid a full reload ?

Reply:
it's generally more secure to make a full reload because the updated information does not appear immediately unless the page is reloaded. do you have any suggestion regarding this?